server/src/main/java/org/opensearch/action/ProtobufActionListenerResponseHandler.java

```diff
@@ -0,0 +1,79 @@
+/*
+* SPDX-License-Identifier: Apache-2.0
+*
+* The OpenSearch Contributors require contributions made to
+* this file be licensed under the Apache-2.0 license or a
+* compatible open source license.
+*/
+
+package org.opensearch.action;
+
+import org.opensearch.core.action.ActionListener;
+import org.opensearch.core.common.io.stream.BytesWriteable;
+import org.opensearch.core.common.io.stream.StreamInput;
+import org.opensearch.core.transport.TransportResponse;
+import org.opensearch.threadpool.ThreadPool;
+import org.opensearch.transport.TransportException;
+import org.opensearch.transport.TransportResponseHandler;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Objects;
+
+/**
+ * A simple base class for action response listeners, defaulting to using the SAME executor (as its
+* very common on response handlers).
+*
+* @opensearch.api
+*/
+public class ProtobufActionListenerResponseHandler<Response extends TransportResponse> implements TransportResponseHandler<Response> {
+
+    private final ActionListener<? super Response> listener;
+    private final BytesWriteable.Reader<Response> reader;
+    private final String executor;
+
+    public ProtobufActionListenerResponseHandler(
+        ActionListener<? super Response> listener,
+        BytesWriteable.Reader<Response> reader,
+        String executor
+    ) {
+        this.listener = Objects.requireNonNull(listener);
+        this.reader = Objects.requireNonNull(reader);
+        this.executor = Objects.requireNonNull(executor);
+    }
+
+    public ProtobufActionListenerResponseHandler(ActionListener<? super Response> listener, BytesWriteable.Reader<Response> reader) {
+        this(listener, reader, ThreadPool.Names.SAME);
+    }
+
+    @Override
+    public void handleResponse(Response response) {
+        listener.onResponse(response);
+    }
+
+    @Override
+    public void handleException(TransportException e) {
+        listener.onFailure(e);
+    }
+
+    @Override
+    public String executor() {
+        return executor;
+    }
+
+    @Override
+    public String toString() {
+        return super.toString() + "/" + listener;
+    }
+
+    @Override
+    public Response read(StreamInput in) throws IOException {
+        // TODO Auto-generated method stub
```

Comment:
Leftovers?

Reply:
This should be here since it is for the ProtobufActionListenerResponseHandler and not the normal handlers for which I wrote the default unimplemented implementation.

Reply:
I mean `// TODO Auto-generated method stub` probably is not needed

Reply:
Oh 🤦🏼‍♀️